packages/components/molecules/f-restaurant-card/src/components/subcomponents/DeliveryTimeMeta/_tests/DeliveryTimeMeta.test.js

```diff
@@ -0,0 +1,185 @@
+import { shallowMount, mount, config } from '@vue/test-utils';
+import sut from '../DeliveryTimeMeta.vue';
+
+
+describe('DeliveryTimeMeta', () => {
+    it('should be defined', () => {
+        // arrange
+        const propsData = {
+            eta: '20 - 25 min'
+        };
+
+        // act
+        const wrapper = shallowMount(sut, { propsData });
+
+        // assert
+        expect(wrapper.exists()).toBe(true);
+    });
+    it('should show eta text when data is present', () => {
+        // arrange
+        const propsData = {
+            eta: 'foo',
+            distance: 'bar',
+            address: 'baz'
+        };
+
+        // act
+        const wrapper = mount(sut, { propsData });
+
+        const visibleText = wrapper.text();
+        const etaExists = wrapper.find('[data-test-id="delivery-meta-eta"]').exists();
+
+        // assert
+        expect(visibleText).toBe('foo');
+        expect(etaExists).toBe(true);
+    });
+
+    it('should fallback to distance when eta data is missing', () => {
+        // arrange
+        const propsData = {
+            distance: 'bar',
+            address: 'baz'
+        };
+
+        // act
+        const wrapper = mount(sut, { propsData });
+
+        const visibleText = wrapper.text();
+        const distanceExists = wrapper.find('[data-test-id="delivery-meta-distance"]').exists();
+
+        // assert
+        expect(visibleText).toBe('bar');
+        expect(distanceExists).toBe(true);
+    });
+
+    it('should fallback to address when eta and distance data is missing', () => {
+        // arrange
+        const propsData = {
+            address: 'baz'
+        };
+
+        // act
+        const wrapper = mount(sut, { propsData });
+
+        const visibleText = wrapper.text();
+        const addressExists = wrapper.find('[data-test-id="delivery-meta-address"]').exists();
+
+        // assert
+        expect(visibleText).toBe('baz');
```

Comment:
we should probably test only the desired variant is rendered and the other 2 are not :)

Reply:
will be fixed forward in the next PR